File: 0069-sqrtx/0069-sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        if (x == 0){
            return 0;
        }
        long long int i = 0;

        int left = 1;
        int right = x;

        while(left <= right){
            int mid = left + (right-left)/2;
            if(x/mid == mid){
                return mid;
            }
            else if (x/mid > mid){
                left = mid+1;
            }
            else{
                right = mid-1;
            }
        }
        return right;
    }
};

File: 0069-sqrtx/0069-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        left = 0
        right = x

        while left <= right:
            mid = (right + left) // 2

            if  mid * mid > x:
                right = mid - 1

            else:
                left = mid + 1

        return right
        